File: Backend/academic-core-Service/src/main/java/com/unisystem/academic_core_service/domain/model/Course.java
package com.unisystem.academic_core_service.domain.model;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDate;

@Getter
@Setter
public class Course {
    private Long id;
    private String name;
    private String courseCode;
    private String description;
    private LocalDate startDate;
    private LocalDate endDate;
    private LocalDate createdAt;
    private int credits;
    private int maxStudents;
    private int enrolledCount;
    private Long departmentId;
    private Long teacherId;

    public Course() {

    }
    public boolean isFull() {
        return enrolledCount >= maxStudents;
    }
    
    public void enrollStudent() {
        if (isFull()) {
            throw new RuntimeException("Course is full");
        }
        enrolledCount++;
    }
    
    public void unenrollStudent() {
        enrolledCount--;
    }


}

File: FrontEnd/my-app/src/Interfaces/course.ts
import type { EnrolledCourseResponse } from "./enrolledCourse";

export interface course{
    id:number;
    name:string;
    description:string;
    department:string;
    courseCode:string;
    startDate:Date;
    endDate:Date;
    teacherName:string;
    credits:number;
    maxStudents:number;
    enrolledStudents:number;
    enrolledCount?: number;
    teacherUserName?: string;
    creditHours?: number;
}

export interface CourseRequest{
    name:string;
    description:string;
    courseCode:string;
    startDate:string;
    endDate:string;
    departmentName:string;
    userId:number;
    creditHours:number;
    maxStudents:number;
}

export interface CourseSidebar {
courseCode:string;
creditHours:number;
credits?: number;
startDate:Date;
endDate:Date;
enrolledStudents:number;
name:string;
teacherUserName:string;
teacherName?: string;
}